packages/server-side-render/src/server-side-render.js

```diff
@@ -20,148 +21,124 @@ export function rendererPath( block, attributes = null, urlQueryArgs = {} ) {
 	} );
 }
 
-export class ServerSideRender extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			response: null,
-		};
-	}
+function DefaultEmptyResponsePlaceholder( { className } ) {
+	return (
+		<Placeholder className={ className }>
+			{ __( 'Block rendered as empty.' ) }
+		</Placeholder>
+	);
+}
 
-	componentDidMount() {
-		this.isStillMounted = true;
-		this.fetch( this.props );
-		// Only debounce once the initial fetch occurs to ensure that the first
-		// renders show data as soon as possible.
-		this.fetch = debounce( this.fetch, 500 );
-	}
+function DefaultErrorResponsePlaceholder( { response, className } ) {
+	const errorMessage = sprintf(
+		// translators: %s: error message describing the problem
+		__( 'Error loading block: %s' ),
+		response.errorMsg
+	);
+	return <Placeholder className={ className }>{ errorMessage }</Placeholder>;
+}
 
-	componentWillUnmount() {
-		this.isStillMounted = false;
-	}
+function DefaultLoadingResponsePlaceholder( { className } ) {
+	return (
+		<Placeholder className={ className }>
+			<Spinner />
+		</Placeholder>
+	);
+}
 
-	componentDidUpdate( prevProps ) {
-		if ( ! isEqual( prevProps, this.props ) ) {
-			this.fetch( this.props );
-		}
-	}
+export default function ServerSideRender( props ) {
+	const {
+		attributes,
+		block,
+		className,
+		httpMethod = 'GET',
+		urlQueryArgs,
+		EmptyResponsePlaceholder = DefaultEmptyResponsePlaceholder,
+		ErrorResponsePlaceholder = DefaultErrorResponsePlaceholder,
+		LoadingResponsePlaceholder = DefaultLoadingResponsePlaceholder,
+	} = props;
+
+	const isMountedRef = useRef( true );
+	const fetchRequestRef = useRef();
+	const [ response, setResponse ] = useState( null );
+	const prevProps = usePrevious( props );
 
-	fetch( props ) {
-		if ( ! this.isStillMounted ) {
+	function fetch() {
```

Comment:
I would refrain from using the function name `fetch` because it is already available on the global window object.

Reply:
Good point. Addressed. :+1: